Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm

```diff
@@ -140,54 +141,106 @@ static void resizeScene(UIWindowScene *scene, CGSize size, CompletionHandler<voi
     void applyTo(WKWebView* webView)
     {
         [webView _setPageScale:_savedPageScale withOrigin:CGPointMake(0, 0)];
-        [webView _setObscuredInsets:_savedObscuredInsets];
-        [[webView scrollView] setContentInset:_savedEdgeInset];
-        [[webView scrollView] setContentOffset:_savedContentOffset];
-        [[webView scrollView] setScrollIndicatorInsets:_savedScrollIndicatorInsets];
+        if (_savedHaveSetObscuredInsets)
+            webView._obscuredInsets = _savedObscuredInsets;
+        else
+            [webView _resetObscuredInsets];
+
+        auto* scrollView = (WKScrollView *)[webView scrollView];
+        if (_savedContentInsetWasExternallyOverridden)
+            scrollView.contentInset = _savedEdgeInset;
+        else
+            [scrollView _resetContentInset];
+
+        scrollView.contentOffset = _savedContentOffset;
+        scrollView.scrollIndicatorInsets = _savedScrollIndicatorInsets;
+
+#if !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
+        if (_savedContentInsetAdjustmentBehaviorWasExternallyOverridden)
+            scrollView.contentInsetAdjustmentBehavior = _savedContentInsetAdjustmentBehavior;
+        else
+            [webView _resetScrollViewInsetAdjustmentBehavior];
+#endif
+
+        if (_savedHaveSetUnobscuredSafeAreaInsets)
+            webView._unobscuredSafeAreaInsets = _savedUnobscuredSafeAreaInsets;
+        else
+            [webView _resetUnobscuredSafeAreaInsets];
+
+        if (_savedHasOverriddenLayoutParameters && _savedMinimumUnobscuredSizeOverride && _savedMaximumUnobscuredSizeOverride)
+            [webView _overrideLayoutParametersWithMinimumLayoutSize:*_savedMinimumUnobscuredSizeOverride maximumUnobscuredSizeOverride:*_savedMaximumUnobscuredSizeOverride];
+        else
+            [webView _clearOverrideLayoutParameters];
+
         if (auto page = webView._page) {
             page->setTopContentInset(_savedTopContentInset);
             page->setForceAlwaysUserScalable(_savedForceAlwaysUserScalable);
         }
         [webView _setViewScale:_savedViewScale];
-        [[webView scrollView] setZoomScale:_savedZoomScale];
-        webView.scrollView.minimumZoomScale = _savedMinimumZoomScale;
-        webView.scrollView.maximumZoomScale = _savedMaximumZoomScale;
-        webView.scrollView.bouncesZoom = _savedBouncesZoom;
+        scrollView.zoomScale = _savedZoomScale;
+        scrollView.minimumZoomScale = _savedMinimumZoomScale;
+        scrollView.maximumZoomScale = _savedMaximumZoomScale;
+        scrollView.bouncesZoom = _savedBouncesZoom;
         webView._minimumEffectiveDeviceWidth = _savedMinimumEffectiveDeviceWidth;
     }
 
     void store(WKWebView* webView)
     {
-        _savedPageScale = [webView _pageScale];
-        _savedObscuredInsets = [webView _obscuredInsets];
-        _savedEdgeInset = [[webView scrollView] contentInset];
-        _savedContentOffset = [[webView scrollView] contentOffset];
+        auto* scrollView = (WKScrollView *)[webView scrollView];
```

Comment:
```suggestion
        auto *scrollView = [webView _wkScrollView];
```

Nit - there's an internal helper that already grabs the scroll view as a `WKScrollView`.